### PR TITLE
Make broad use of PREFIX_ID

### PR DESCRIPTION
**Pull request by mdellweg**
[noissue]

### CODE REVIEW

File: pulpcore/cli/core/context.py [hunk 1]
```diff
@@ -379,24 +321,20 @@ class PulpTaskGroupContext(PulpEntityContext):
...
     HREF = "upload_href"
-    LIST_ID = "uploads_list"
-    READ_ID = "uploads_read"
-    CREATE_ID = "uploads_create"
-    UPDATE_ID = "uploads_update"
-    DELETE_ID = "uploads_delete"
-    COMMIT_ID: ClassVar[str] = "uploads_commit"
+    ID_PREFIX = "uploads"
+    # TODO: Not so nice
+    PARTIAL_UPDATE_ID = "uploads_update"
```
Comment (mdellweg) [inline pulpcore/cli/core/context.py:333]: This works, but i am unsure whether it's a sign that we should not use `update()`, but implement a `upload_chunk()` on this context.

File: pulpcore/cli/common/debug.py [hunk 2]
```diff
@@ -48,7 +48,7 @@ def task_summary(pulp_ctx: PulpContext) -> None:
     result = {}
     for state in TASK_STATES:
         payload = {"limit": 1, "state": state}
-        answer = pulp_ctx.call(PulpTaskContext.LIST_ID, parameters=payload)
+        answer = PulpTaskContext(pulp_ctx).call("list", parameters=payload)
```
[the remark this answers is not in the document]
Comment (mdellweg) [reply pulpcore/cli/common/debug.py:51]: `def summary(self)` it is.

File: pulpcore/cli/core/context.py (hunk 1 above)
Comment (mdellweg) [reply pulpcore/cli/core/context.py]: `upload_chunk` it is.

File: pulpcore/cli/core/context.py [hunk 3]
```diff
@@ -369,34 +308,56 @@ def purge(self, finished_before: Optional[str], states: Optional[List[str]]) ->
...
         )
 
+    def summary(self) -> Dict[str, int]:
+        task_states = ["waiting", "skipped", "running", "completed", "failed", "canceled"]
+        if self.pulp_ctx.has_plugin(PluginRequirement("core", min="3.14")):
+            task_states.append("canceling")
+        result = {}
+        for state in task_states:
+            payload = {"limit": 1, "state": state}
+            result[state] = self.call("list", parameters=payload)["count"]
```
Comment (ggainey) [inline pulpcore/cli/core/context.py:323]: Oh, *that* is clever - :100:

File: pyproject.toml [hunk 4]
```diff
@@ -51,7 +51,8 @@ line_length = 100
 
 [tool.pytest.ini_options]
 markers = [
-  "script: Marks tests provided as shell scripts",
+  "script: tests provided as shell scripts",
+  "help_page: tests that render help pages",
```
Comment (ggainey) [inline pyproject.toml:55]: Does this belong in this PR? Or should it be its own thing?

**Review by ggainey — COMMENTED** [review]
A question for clarity: will plugins you can't refactor here (e.g., deb-cli) continue working as-is? Or does this break them?

File: pyproject.toml (hunk 4 above)
Comment (mdellweg) [reply pyproject.toml:55]: It's not enough for me to bother making a separate PR for it...

File: pulpcore/cli/common/context.py [hunk 5]
```diff
@@ -512,7 +512,7 @@ def update(
         if parameters:
             _parameters.update(parameters)
         return self.call(
-            "update",
+            "partial_update",
```
Comment (mdellweg) [inline pulpcore/cli/common/context.py:515]: This is the only change with a risk to break anyone. But i don't suspect anyone to use `update` aka `PUT` for any good reason.
(Yes our upload api uses it to create chunks. You'd kind of expect a POST call for that, no? And that weird case is covered explicitely in this PR.)
...is what i wanted to say. But thinking again, this is breaking all contexts using UPDATE_ID today and defining neither PARTIAL_UPDATE_ID or nor ID_PREFIX.

Comment (gerrod3) [reply pulpcore/cli/common/context.py:515]: I think that is a fine assumption to make. I looked at the other two plugins (deb & ostree) and they both use partial_update, so I think we are good.

Comment (mdellweg) [reply pulpcore/cli/common/context.py:519]: I added a workaround. Please look again. @ggainey

Comment (gerrod3) [reply pulpcore/cli/common/context.py:519]: Yes, you are right. I had to go look at #444 to understand how this works.

File: pulpcore/cli/common/context.py [hunk 6]
```diff
@@ -508,11 +508,15 @@ def update(
         uploads: Optional[Dict[str, Any]] = None,
         non_blocking: bool = False,
     ) -> Any:
+        # Workaround for plugins that do not have ID_PREFIX in place
```
Comment (ggainey) [inline pulpcore/cli/common/context.py:511]: :+1:

**Review by ggainey — APPROVED** [review]
"Assuming" that we know the current state of things no longer breaks pulp_deb, looks good to me!

**Review by gerrod3 — APPROVED** [review]
Looks good. Linter is complaining about the workaround, but after that's fixed LGTM.

**Pull request merged**